Compilation failure on Pi4

### Platform

Raspberry Pi

### Compiler and build tool versions

gcc (GCC) 12.1.0

### Operating system version

Arch Linux ARM

### Build commands used

./b.sh --rpi64

### What happens

build fails with the following error, 
**however** ,  ‘-flax-vector-conversions’ does indeed work (compiled and built it that way). Maybe want to add said compiler flag to the cmake toolchain file

```
/store/ppsspp/Core/HW/MediaEngine.cpp: In function ‘void writeVideoLineRGBA(void*, const void*, int)’:
/store/ppsspp/Core/HW/MediaEngine.cpp:773:48: note: use ‘-flax-vector-conversions’ to permit conversions between vectors with differing element types or numbers of subparts
  773 |         int32x4_t mask = vdupq_n_u32(0x00FFFFFF);
      |                                                ^
/store/ppsspp/Core/HW/MediaEngine.cpp:773:37: error: cannot convert ‘uint32x4_t’ to ‘int32x4_t’ in initialization
  773 |         int32x4_t mask = vdupq_n_u32(0x00FFFFFF);
      |                          ~~~~~~~~~~~^~~~~~~~~~~~
      |                                     |
      |                                     uint32x4_t
/store/ppsspp/Core/HW/MediaEngine.cpp:775:63: error: cannot convert ‘int32x4_t’ to ‘uint32x4_t’
  775 |                 int32x4_t pixels1 = vandq_u32(vld1q_u32(src), mask);
      |                                                               ^~~~
      |                                                               |
      |                                                               int32x4_t
In file included from /store/ppsspp/GPU/GPUState.h:35,
                 from /store/ppsspp/Core/HW/MediaEngine.cpp:25:
/usr/lib/gcc/aarch64-unknown-linux-gnu/12.1.0/include/arm_neon.h:1134:39: note:   initializing argument 2 of ‘uint32x4_t vandq_u32(uint32x4_t, uint32x4_t)’
 1134 | vandq_u32 (uint32x4_t __a, uint32x4_t __b)
      |                            ~~~~~~~~~~~^~~
/store/ppsspp/Core/HW/MediaEngine.cpp:776:67: error: cannot convert ‘int32x4_t’ to ‘uint32x4_t’
  776 |                 int32x4_t pixels2 = vandq_u32(vld1q_u32(src + 4), mask);
      |                                                                   ^~~~
      |                                                                   |
      |                                                                   int32x4_t
/usr/lib/gcc/aarch64-unknown-linux-gnu/12.1.0/include/arm_neon.h:1134:39: note:   initializing argument 2 of ‘uint32x4_t vandq_u32(uint32x4_t, uint32x4_t)’
 1134 | vandq_u32 (uint32x4_t __a, uint32x4_t __b)
      |                            ~~~~~~~~~~~^~~
/store/ppsspp/Core/HW/MediaEngine.cpp:777:33: error: cannot convert ‘int32x4_t’ to ‘uint32x4_t’
  777 |                 vst1q_u32(dest, pixels1);
      |                                 ^~~~~~~
      |                                 |
      |                                 int32x4_t
/usr/lib/gcc/aarch64-unknown-linux-gnu/12.1.0/include/arm_neon.h:24783:38: note:   initializing argument 2 of ‘void vst1q_u32(uint32_t*, uint32x4_t)’
24783 | vst1q_u32 (uint32_t *__a, uint32x4_t __b)
      |                           ~~~~~~~~~~~^~~
/store/ppsspp/Core/HW/MediaEngine.cpp:778:37: error: cannot convert ‘int32x4_t’ to ‘uint32x4_t’
  778 |                 vst1q_u32(dest + 4, pixels2);
      |                                     ^~~~~~~
      |                                     |
      |                                     int32x4_t
/usr/lib/gcc/aarch64-unknown-linux-gnu/12.1.0/include/arm_neon.h:24783:38: note:   initializing argument 2 of ‘void vst1q_u32(uint32_t*, uint32x4_t)’
24783 | vst1q_u32 (uint32_t *__a, uint32x4_t __b)
      |                           ~~~~~~~~~~~^~~

```

### PPSSPP version affected

v1.16.6-820-gb86e190e5

### Last working version

_No response_

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] Make sure to run `git submodule update --init --recursive` before building.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try deleting the build directory and running the build again.
- [X] Check GitHub Actions, which builds every merge and PR.
- [X] Include logs and help us reproduce.

--- a/Common/Math/CrossSIMD.h
+++ b/Common/Math/CrossSIMD.h
@@ -10,6 +10,13 @@
 
 #include "stdint.h"
 
+#ifdef __clang__
+// Weird how you can't just use #pragma in a macro.
+#define DO_NOT_VECTORIZE_LOOP _Pragma("clang loop vectorize(disable)")
+#else
+#define DO_NOT_VECTORIZE_LOOP
+#endif
+
 #if PPSSPP_ARCH(SSE2)
 #include <emmintrin.h>
 #endif

--- a/Core/HW/MediaEngine.cpp
+++ b/Core/HW/MediaEngine.cpp
@@ -16,6 +16,7 @@
 // https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
 
 #include "Common/Serialize/SerializeFuncs.h"
+#include "Common/Math/CrossSIMD.h"
 #include "Core/Config.h"
 #include "Core/Debugger/MemBlockInfo.h"
 #include "Core/HW/MediaEngine.h"
@@ -781,10 +782,10 @@ inline void writeVideoLineRGBA(void *destp, const void *srcp, int width) {
 		count -= 8;
 	}
 #elif PPSSPP_ARCH(ARM_NEON)
-	int32x4_t mask = vdupq_n_u32(0x00FFFFFF);
+	uint32x4_t mask = vdupq_n_u32(0x00FFFFFF);
 	while (count >= 8) {
-		int32x4_t pixels1 = vandq_u32(vld1q_u32(src), mask);
-		int32x4_t pixels2 = vandq_u32(vld1q_u32(src + 4), mask);
+		uint32x4_t pixels1 = vandq_u32(vld1q_u32(src), mask);
+		uint32x4_t pixels2 = vandq_u32(vld1q_u32(src + 4), mask);
 		vst1q_u32(dest, pixels1);
 		vst1q_u32(dest + 4, pixels2);
 		src += 8;
@@ -793,6 +794,7 @@ inline void writeVideoLineRGBA(void *destp, const void *srcp, int width) {
 	}
 #endif
 	const u32 mask32 = 0x00FFFFFF;
+	DO_NOT_VECTORIZE_LOOP
 	while (count--) {
 		*dest++ = *src++ & mask32;
 	}

--- a/GPU/Common/TextureDecoder.cpp
+++ b/GPU/Common/TextureDecoder.cpp
@@ -23,6 +23,7 @@
 #include "Common/Data/Convert/ColorConv.h"
 #include "Common/CPUDetect.h"
 #include "Common/Log.h"
+#include "Common/Math/CrossSIMD.h"
 
 #include "GPU/GPU.h"
 #include "GPU/GPUState.h"
@@ -39,13 +40,6 @@
 #else
 #include <arm_neon.h>
 #endif
-#endif
-
-#ifdef __clang__
-// Weird how you can't just use #pragma in a macro.
-#define DO_NOT_VECTORIZE_LOOP _Pragma("clang loop vectorize(disable)")
-#else
-#define DO_NOT_VECTORIZE_LOOP
 #endif
 
 const u8 textureBitsPerPixel[16] = {

